packages/core/test/internal/query-executor.test.ts

```diff
@@ -0,0 +1,390 @@
+  function createExecutor ({
+    mockSessionClose,
+    mockSessionExecuteRead,
+    mockSessionExecuteWrite
+  }: {
+    mockSessionClose?: (spy: jest.SpyInstance<Promise<void>>) => void
+    mockSessionExecuteRead?: (spy: jest.SpyInstance<Promise<unknown>, [transactionWork: ManagedTransactionWork<unknown>, transactionConfig?: TransactionConfig | undefined]>) => void
+    mockSessionExecuteWrite?: (spy: jest.SpyInstance<Promise<unknown>, [transactionWork: ManagedTransactionWork<unknown>, transactionConfig?: TransactionConfig | undefined]>) => void
+  } = { }): {
+      queryExecutor: QueryExecutor
+      sessionsCreated: Array<{
+        session: Session
+        spyOnExecuteRead: jest.SpyInstance<any>
+        spyOnExecuteWrite: jest.SpyInstance<any>
+        spyOnClose: jest.SpyInstance<Promise<void>>
+
+      }>
+      createSession: jest.Mock<Session, [args: any]>
+    } {
+    const _mockSessionClose = mockSessionClose ?? ((spy) => spy.mockResolvedValue())
+    const _mockSessionExecuteRead = mockSessionExecuteRead ?? ((spy) => spy.mockResolvedValue({}))
+    const _mockSessionExecuteWrite = mockSessionExecuteWrite ?? ((spy) => spy.mockResolvedValue({}))
+
+    const sessionsCreated: Array<{
+      session: Session
+      spyOnExecuteRead: jest.SpyInstance<any>
+      spyOnExecuteWrite: jest.SpyInstance<any>
+      spyOnClose: jest.SpyInstance<Promise<void>>
+
+    }> = []
+    const createSession = jest.fn((args) => {
+      const session = new Session(args)
+      const sessionCreated = {
+        session,
+        spyOnExecuteRead: jest.spyOn(session, 'executeRead'),
+        spyOnExecuteWrite: jest.spyOn(session, 'executeWrite'),
+        spyOnClose: jest.spyOn(session, 'close')
+      }
+      sessionsCreated.push(sessionCreated)
+      _mockSessionExecuteRead(sessionCreated.spyOnExecuteRead)
+      _mockSessionExecuteWrite(sessionCreated.spyOnExecuteWrite)
+      _mockSessionClose(sessionCreated.spyOnClose)
+      return session
+    })
+    const queryExecutor = new QueryExecutor(createSession)
+
+    return {
+      queryExecutor,
+      sessionsCreated,
+      createSession
+    }
+  }
+
+  function createManagedTransaction (): {
+    managedTransaction: ManagedTransaction
+    spyOnRun: jest.SpyInstance<Result, [query: Query, parameters?: any]>
+    resultObservers: ResultStreamObserverMock[]
+    results: Result[]
+  } {
+    const resultObservers: ResultStreamObserverMock[] = []
+    const results: Result[] = []
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const managedTransaction: ManagedTransaction = {
+      run: (query: string, parameters?: any): Result => {
+        const resultObserver = new ResultStreamObserverMock()
+        resultObservers.push(resultObserver)
+        const result = new Result(
+          Promise.resolve(resultObserver),
+          query,
+          parameters
+        )
+        results.push(result)
+        return result
+      }
+    } as ManagedTransaction
```

Comment:
It is because I don't defined the `_run` in the object, which is expected by the `ManagedTransaction`. I solved it by removing the the `: ManagedTransaction` in the definition (which is the preferred way, but it is not possible because of the `_run`).